src/control/Control.Layers.js

```diff
@@ -184,15 +184,34 @@ export var Layers = Control.extend({
 		var section = this._section = DomUtil.create('section', className + '-list');
 
 		var link = this._layersLink = DomUtil.create('a', className + '-toggle', container);
+		var keyboardFocus = false;
 		link.href = '#';
 		link.title = 'Layers';
 		link.setAttribute('role', 'button');
 		DomEvent.on(link, 'click', DomEvent.preventDefault); // prevent link function
-		DomEvent.on(link, 'focus', this.expand, this);
+		DomEvent.on(link, 'focus',
+			() => {
+				keyboardFocus = true;
+			}
+			, this);
+		DomEvent.on(link, 'focusout',
+			() => {
+				keyboardFocus = false;
+			}
+		);
+		DomEvent.on(link, 'keydown', () => {
+			if (keyboardFocus) {
+				this.expand();
+				link.setAttribute('aria-expanded', 'true');
+			}
+		});

```

Comment:
`keydown` is intercepted by both the NVDA and Narrator [screen readers](https://leafletjs.com/examples/accessibility/#test-your-maps), and the control is not toggled. `click` works OTOH.

Reply:
I have addressed these issues in commit [3812689](https://github.com/Leaflet/Leaflet/pull/8202/commits/381268965b3badb0f34ae6eb446660026af2dadd). Let me know if it meets your requirements